### PR TITLE
Typescript for pp display component

### PR DESCRIPTION
**Pull request by nanaya**
- [x] depends on #8295

### CODE REVIEW

File: resources/assets/lib/interfaces/user-statistics-json.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
 // See the LICENCE file in the repository root for full licence text.
 
-import GameMode from 'interfaces/game-mode';
+import GameMode from './game-mode';
```
Comment (notbakaneko) [inline resources/assets/lib/interfaces/user-statistics-json.ts:4]: can revert this change

**Review by notbakaneko — CHANGES_REQUESTED** [review]

File: resources/assets/lib/interfaces/user-statistics-json.ts (hunk 1 above)
Comment (nanaya) [reply resources/assets/lib/interfaces/user-statistics-json.ts:4]: should it though :thinking:

**Review by notbakaneko — APPROVED** [review]
